test/test_cuda.py

```diff
@@ -1108,6 +1108,49 @@ def test_inplace_unary_mem_overlap(self):
     def test_inplace_binary_mem_overlap(self):
         _TestTorchMixin._test_inplace_binary_mem_overlap(self, device='cuda')
 
+    # This test does not work until input/output overlap check is implemented on CUDA.
+    # Remove @unittest.expectedFailure when it's done.
+    @unittest.expectedFailure
```

Comment:
Do all of the following functions have no input-output mem overlap check, or do some of them not have it?

If it is the latter, we should split up this test so that it is clear which functions need to be addressed.

Reply:
addressed